client/components/login/login.js

```diff
@@ -174,31 +179,66 @@ export default class Login extends React.Component {
+                {settings.mobile_phone_verification && input_fields.phone_number && (
+                  <div className="row phone-number">
+                    <label htmlFor="phone-number">
+                      {getText(input_fields.phone_number.label, language)}
+                    </label>
+
+                    {errors.phone_number && (
+                      <div className="error">
+                        <span className="icon">!</span>
+                        <span className="text">
+                          {errors.phone_number}
+                        </span>
+                      </div>
+                    )}
+                    <PhoneInput
+                      name="phone_number"
+                      country={input_fields.phone_number.country}
+                      onlyCountries={input_fields.phone_number.only_countries || []}
+                      preferredCountries={input_fields.phone_number.preferred_countries || []}
+                      excludeCountries={input_fields.phone_number.exclude_countries || []}
+                      value={phone_number}
+                      onChange={value => this.handleChange({ target: { name: "phone_number", value: `+${value}` } })}
+                      placeholder={getText(input_fields.phone_number.placeholder, language)}
+                      enableSearch={Boolean(input_fields.phone_number.enable_search)}
+                      inputProps={{
+                        name: "phone_number",
+                        id: "phone-number",
+                        className: `form-control input ${errors.phone_number ? "error" : ""}`,
+                        required: true,
+                      }}
+                    />
+                  </div>
+                )}
+                {!settings.mobile_phone_verification && (
+                  <div className="row email">
+                    <label htmlFor="email">
+                      {getText(input_fields.email.label, language)}
+                    </label>
+                    {errors.email && (
+                      <div className="error">
+                        <span className="icon">!</span>
+                        <span className="text">
+                          {errors.email}
+                        </span>
+                      </div>
+                    )}
+                    <input
+                      className={`input ${errors.email ? "error" : ""}`}
+                      type={input_fields.email.type}
+                      id="email"
+                      name="email"
+                      value={email}
+                      onChange={this.handleChange}
+                      required
+                      placeholder={getText(input_fields.email.placeholder, language)}
+                      pattern={input_fields.email.pattern}
+                      title={getText(input_fields.email.pattern_description, language)}
+                    />
+                  </div>
+                )}
```

Comment:
I think we could make a method here:
```javascript
{this.renderLoginField()}
```

```javascript
renderLoginField() {
    if (settings.mobile_phone_verification) {
        return this.renderPhoneField();
    }
    return this.renderEmailField();
}
```

See also https://reactjs.org/docs/conditional-rendering.html.